src/server.c

```diff
@@ -3992,14 +3992,13 @@ int processCommand(client *c) {
         return C_OK;
     }
 
-    /* If the server is paused, block the client until
-     * the pause has ended. Replicas are never paused. */
+    /* If the server is paused, block the client until the pause has ended. Replicas never paused.
+     * First if, though looks redundant, optimized for the case that none is paused */
     if (isPausedActions(PAUSE_ACTION_CLIENT_DENYOOM|PAUSE_ACTION_CLIENT_WRITE|PAUSE_ACTION_CLIENT_ALL)) {
         if (!(c->flags & CLIENT_SLAVE) &&
             ((isPausedActions(PAUSE_ACTION_CLIENT_ALL)) ||
-             ((is_may_replicate_command) &&
-              ((isPausedActions(PAUSE_ACTION_CLIENT_WRITE)) ||
-               (isPausedActions(PAUSE_ACTION_CLIENT_DENYOOM) && (is_denyoom_command)))))) {
+             ((is_may_replicate_command) && (isPausedActions(PAUSE_ACTION_CLIENT_WRITE))) ||
+             (isPausedActions(PAUSE_ACTION_CLIENT_DENYOOM) && (is_denyoom_command)))) {
```

Comment:
let's make the order consistent across lines.
```suggestion
             (isPausedActions(PAUSE_ACTION_CLIENT_WRITE) && is_may_replicate_command) ||
             (isPausedActions(PAUSE_ACTION_CLIENT_DENYOOM) && is_denyoom_command))) {
```